packages/opencensus-web-instrumentation-zone/test/test-on-page-interaction.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { RootSpan, Tracer } from '@opencensus/web-core';
+import { OnPageInteractionStopwatch } from '../src/on-page-interaction';
+
+describe('OnPageInteraction', () => {
+  let root: RootSpan;
+  let tracer: Tracer;
+  let interaction: OnPageInteractionStopwatch;
+  let target: HTMLElement;
+
+  describe('Tasks tracking', () => {
+    beforeEach(() => {
+      tracer = Tracer.instance;
+      root = new RootSpan(tracer, { name: 'root1' });
+      target = document.createElement('button');
+      const interactionData = {
+        startLocationHref: location.href,
+        startLocationPath: location.pathname,
+        eventType: 'click',
+        target,
+        rootSpan: root,
+      };
+      interaction = new OnPageInteractionStopwatch(interactionData);
+    });
+    it('should track number of active tasks', () => {
+      interaction.incrementTaskCount();
+      interaction.incrementTaskCount();
+      expect(interaction.getTaskCount()).toBe(2);
+      interaction.decrementTaskCount();
+      expect(interaction.getTaskCount()).toBe(1);
+      expect(interaction.hasRemainingTasks()).toBeTruthy();
+      interaction.decrementTaskCount();
+      expect(interaction.getTaskCount()).toBe(0);
+      // Decrement again to test no decrement if count is 0.
+      interaction.decrementTaskCount();
+      expect(interaction.getTaskCount()).toBe(0);
+      expect(interaction.hasRemainingTasks()).toBeFalsy();
+    });
+  });
+
+  describe('stopAndRecord()', () => {
+    beforeEach(() => {
+      tracer = Tracer.instance;
+      root = new RootSpan(tracer, { name: 'root1' });
+      target = document.createElement('button');
+      const interactionData = {
+        startLocationHref: location.href,
+        startLocationPath: location.pathname,
+        eventType: 'click',
+        target,
+        rootSpan: root,
+      };
+      interaction = new OnPageInteractionStopwatch(interactionData);
+    });
+    it('should stop an on page interaction and end the root span', () => {
+      spyOn(tracer, 'onEndSpan');
+      interaction.stopAndRecord();
+
+      expect(root.attributes['EventType']).toBe('click');
```

Comment:
Should the test cases above for the fake user interactions also test these attributes?

Reply:
Thanks, Added.